rdrf/rdrf/services/io/notifications/email_notification.py

```diff
@@ -233,47 +231,44 @@ def append(self, key, obj):
-        email.recipients = additional_recipients
+        email.recipient_dict = mandatory_recipients
         email.template_data = template_data or {}
         return email.send()
 
 
-def process_notification(reg_code=None, description=None, template_data=None, additional_recipients=None):
-    template_data = template_data or {}
-    notes = EmailNotification.objects.filter(registry__code=reg_code, description=description)
-    has_disabled = False
-    sent_successfully = True
-    for note in notes:
-        if note.disabled:
-            logger.warning("Email %s disabled" % note)
-            has_disabled = True
-            continue
-        send_result = process_given_notification(note, template_data, additional_recipients)
-        sent_successfully = sent_successfully and send_result
-    return sent_successfully, has_disabled
-
-
-def process_notification_with_default(reg_code=None, description=None, template_data=None, default_template=None, default_subject=None, default_recipient=None):
+def process_raw_email(subject, message_template, message_template_data, recipient_list):
+    email_body = message_template.render(Context(message_template_data).flatten())
+
+    return send_mail(subject=subject,
+                     message=email_body,
+                     from_email=settings.DEFAULT_FROM_EMAIL,
+                     recipient_list=recipient_list,
+                     html_message=email_body)
+
+
+def process_notification(reg_code=None, description=None, template_data=None, mandatory_recipients=None, default_template=None, default_subject=None):
     template_data = template_data or {}
+    mandatory_recipients = mandatory_recipients or {}
     notes = EmailNotification.objects.filter(registry__code=reg_code, description=description)
-
     if notes:
-        return process_notification(reg_code, description, template_data, default_recipient)
+        has_disabled = False
+        sent_successfully = True
+        for note in notes:
+            if note.disabled:
+                logger.warning("Email %s disabled" % note)
+                has_disabled = True
+                continue
+            send_result = process_given_notification(note, template_data, mandatory_recipients)
+            sent_successfully = sent_successfully and send_result
+        return sent_successfully, has_disabled
     elif default_template:
         logger.debug('Registry notification not found, using default template')
-        email_body = default_template.render(Context(template_data).flatten())
-
-        send_mail(subject=default_subject,
-                  message=email_body,
-                  from_email=settings.DEFAULT_FROM_EMAIL,
-                  recipient_list=default_recipient,
-                  html_message=email_body)
-
-        return True, False  # has sent successfully, wasn't disabled.
+        num_sent_emails = process_raw_email(default_subject, default_template, template_data, mandatory_recipients)
```

Comment:
Using duck typing to treat a dict like a list here is a bit iffy to me.

```suggestion
        recipient_list = list(mandatory_recipients.keys())
        num_sent_emails = process_raw_email(default_subject, default_template, template_data, recipient_list)
```

Reply:
It was just missed in my refactoring. Will fix

Reply:
Committed fix